src/main/java/net/sf/jabref/logic/autosaveandbackup/AutosaveManager.java

```diff
@@ -0,0 +1,94 @@
+package net.sf.jabref.logic.autosaveandbackup;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import net.sf.jabref.model.database.BibDatabaseContext;
+import net.sf.jabref.model.database.event.AutosaveEvent;
+import net.sf.jabref.model.database.event.BibDatabaseContextChangedEvent;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Saves the given {@link BibDatabaseContext} on every {@link BibDatabaseContextChangedEvent} by posting a new {@link AutosaveEvent}.
+ * An intelligent {@link ExecutorService} with a {@link BlockingQueue} prevents a high load while saving and rejects all redundant save tasks.
+ */
+public class AutosaveManager {
+
+    private static final Log LOGGER = LogFactory.getLog(AutosaveManager.class);
+
+    private static Set<AutosaveManager> runningInstances = new HashSet<>();
+
+    private final BibDatabaseContext bibDatabaseContext;
+    private final BlockingQueue<Runnable> workerQueue;
+    private final ExecutorService executor;
+    private final EventBus eventBus;
+
+
+    private AutosaveManager(BibDatabaseContext bibDatabaseContext) {
+        this.bibDatabaseContext = bibDatabaseContext;
+        this.workerQueue = new ArrayBlockingQueue<>(1);
+        this.executor = new ThreadPoolExecutor(1, 1, 0, TimeUnit.SECONDS, workerQueue);
+        this.eventBus = new EventBus();
+    }
+
+    @Subscribe
+    public synchronized void listen(@SuppressWarnings("unused") BibDatabaseContextChangedEvent event) {
```

Comment:
What is the purpose of this class? It looks like it just reposts the event, right? Why can't one directly subscribe to the BibDatabaseContextChangedEvents of the BibDatabaseContext?

But this class might be a solution for the above issue with setEncoding and not firing the event. You pause the AutosaveManager once you begin with writing the file so that changes (SaveActions?) during the writing process don't trigger an additional writing.